### ISSUE TITLE
Use MultivariatePolynomials

### ISSUE THREAD

**Issue by blegat** [JuliaPolyhedra/Polyhedra.jl#43]
Now that MultivariatePolynomials is lightweight thanks to https://github.com/blegat/MultivariatePolynomials.jl/pull/41, we could add it to the requirement of Polyhedra and makes the `HRepresentation` and `Polyhedron` types `BasicSemialgebraicSet` subtypes.
We could define a `Polyhedra.Variable` type with a `Polyhedra.@polyvar` macro and an `AffineTerm` and an `AffinePolynomial` types.
This way, one could define an `HRepresentation` with
```
@polyvar x y
@set x + y == 1 && x >= 0 && y >= 0
```
The `inequalities` and `equalities` iterators would return `AffinePolynomial`s.
@rdeits what do you think ?

**Comment by chriscoey** [comment]
If this would make it easier to work with Newton polytopes for SOS optimization than I'm in favor!

**Comment by blegat** [comment]
I don't think it would help with Newton polytopes, it just gives a nice interface to build polytopes similar to SemialgebraicSets, I don't think it is a high priority.
